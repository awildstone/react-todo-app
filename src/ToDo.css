button {
	margin-left: 10px;
	/* background-color: lightseagreen; */
	padding: 5px;
	order: 1px solid black;
	border-radius: 5px;
}

.delete {
	background-color: orangered;
}

.save {
	background-color: seagreen;
}
.update {
	background-color: lightseagreen;
}

.edit {
	background-color: goldenrod;
}

.buttons {
	text-align: center;
}

.task {
	margin-left: auto;
	margin-right: auto;
	width: 400px;
	text-align: center;
	margin-bottom: 10px;
	padding: 10px;
	border: 1px solid black;
	border-radius: 5px;
	background-color: lightgray;
	overflow-wrap: break-word;
}

.task-form {
	margin-left: auto;
	margin-right: auto;
	width: 400px;
	text-align: left;
	margin-bottom: 10px;
	padding: 10px;
	border: 1px solid black;
	border-radius: 5px;
	background-color: lightgray;
}
