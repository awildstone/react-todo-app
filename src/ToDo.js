import React, { useState } from 'react';
import './ToDo.css';

const ToDo = ({id, task, removeToDo, editToDo }) => {
    const [editTask, setEditTask] = useState(task);
    const [isEditing, setIsEditing] = useState(false);
    const [isComplete, setIsComplete] = useState(false);

    const toggleEdit = () => {
        setIsEditing(edit => !edit);
    };

    const handleEdit = (evt) => {
        setEditTask(evt.target.value);
    };

    const handleUpdate = (evt) => {
        evt.preventDefault();
        editToDo(id, editTask);
        setIsEditing(false);
    };

    const handleDelete = () => {
        removeToDo(id);
    }

    const toggleComplete = () => {
        setIsComplete(done => !done);
    }

    let strikeTask = <span style={ {textDecoration: 'line-through'} }>{editTask}</span>;

    let toDoJsx = ( 
        <div className="task" data-testid={id}>
            {isComplete ? strikeTask : editTask}
            <div className="buttons">
                <button className="update" onClick={toggleComplete}>Mark as completed</button>
                <button className="edit" onClick={toggleEdit}>Edit</button>
                <button className="delete" onClick={handleDelete}>X</button>
            </div>
        </div>
    );

    if (isEditing) {
        toDoJsx = (
            <div className="task-form">
                <form onSubmit={handleUpdate}>
                    <input type="text" value={editTask} onChange={handleEdit} />
                    <button className="save">Update</button>
                </form>
            </div>
      )
    }

    return toDoJsx;
}

export default ToDo;