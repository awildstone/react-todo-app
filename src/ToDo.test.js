import React from 'react';
import { render } from '@testing-library/react';
import ToDo from './ToDo';

it('renders a ToDo without crashing', () => {
    render(<ToDo />);
});

it('matches a ToDo snapshot', () => {
    const { asFragment } = render(<ToDo />);
    expect(asFragment()).toMatchSnapshot();
});