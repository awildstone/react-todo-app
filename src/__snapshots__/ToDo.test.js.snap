// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a ToDo snapshot 1`] = `
<DocumentFragment>
  <div
    class="task"
  >
    <div
      class="buttons"
    >
      <button
        class="update"
      >
        Mark as completed
      </button>
      <button
        class="edit"
      >
        Edit
      </button>
      <button
        class="delete"
      >
        X
      </button>
    </div>
  </div>
</DocumentFragment>
`;
